libs/orders/src/lib/order-hooks/use-order-validation.tsx

```diff
@@ -28,33 +28,112 @@ export const useOrderValidation = ({
 }: ValidationProps) => {
   const { keypair } = useVegaWallet();
 
-  const invalidText = useMemo(() => {
+  const { message, isDisabled } = useMemo(() => {
     if (!keypair) {
-      return t('No public key selected');
+      return { message: t('No public key selected'), isDisabled: true };
     }
 
     if (keypair.tainted) {
-      return t('Selected public key has been tainted');
+      return {
+        isDisabled: true,
+        message: t('Selected public key has been tainted'),
+      };
+    }
+
+    if (
```

Comment:
@campbellssource @PeteWilliams @jtsang586 

it would be great if I can have your review on this file.

Reply:
Also @candida-d